### ISSUE TITLE
Webpack 5 module federation: output.publicPath only works if an url

### ISSUE THREAD

**Issue by raix** [webpack/webpack#11778]
# Bug report

**What is the current behavior?**
It fails to load vendor chunks:
![image](https://user-images.githubusercontent.com/1136718/96773771-5043ef00-13e5-11eb-958b-de6463c22969.png)
*the files are served without the publicPath on the root on app1 but correctly on app2 - just not loaded - it would never make sense that app1 and app2 are mounted on the same publicPath so somethings is missing for this to work*

**If the current behavior is a bug, please provide the steps to reproduce.**
* clone https://github.com/module-federation/module-federation-examples/tree/master/typescript
in app1:
* set devServer `publicPath: "/test/",`
* set output `publicPath: "/test/",` (`publicPath: "http://localhost:3002/test/",` works...)
in app2:
* update remote entry to: `app2: "app2@http://localhost:3002/test/remoteEntry.js",`

**What is the expected behavior?**
That vendor chunks load:
![image](https://user-images.githubusercontent.com/1136718/96773977-9e58f280-13e5-11eb-8041-9b4dd9ebf5d2.png)

I haven't yet tested if using a public path instead of an url in the remotes config works - but I would expected it to:
```js
      remotes: {
        app1: "app1@/test/remoteEntry.js",
      },
```
*(when hosted on the same domain - otherwise having a `publicPath` would not make sense right?)*

**Other relevant information:**
webpack version:  5.1.3
Node.js version: 14.13.1
Operating System: mac OS Catalina
Additional tools:

**Comment by evilebottnawi** [comment]
dev server doesn't support module federation

**Comment by evilebottnawi** [comment]
Other related issue https://github.com/webpack/webpack/issues/11182

**Comment by ScriptedAlchemy** [comment]
Yeah there's workarounds. Contact me or chat on the other thread.

**Comment by raix** [comment]
Thanks @ScriptedAlchemy I might reach out 👍  (not sure if the solution is to manually configure the client - examples I've seen)

Fortunately we proxy the front-end into a development cluster locally so I might be able to use publicPath as url locally - I did a production build test and things seemed to work as expected (so I'm happy about that)

@evilebottnawi I'm not sure if it's already being looked into or if you need help to track down a solution for it? *(I could not find any related issues or pull-requests on webpack-dev-server - I could have missed it)*

**Comment by skyrex-mark** [comment]
Hello @raix , out of curiosity, have you figured out a way to do it without the proxy? I am on the same boat as you. My team has a proxy service that can do the routing correctly, however, it will break without the proxy which makes the dev experience kinda bad.

**Comment by rajeshbabu-oviva** [comment]
@ScriptedAlchemy Hey Zack is there a work around or fix for this, I am still facing this and hardcoding the publicPath on the remote app fixes it, but it could be problematic and annoying for us in the future.

**Comment by XiaofengXie16** [comment]
> @ScriptedAlchemy Hey Zack is there a work around or fix for this, I am still facing this and hardcoding the publicPath on the remote app fixes it, but it could be problematic and annoying for us in the future.

This may be want you needed.
https://github.com/module-federation/module-federation-examples/issues/566
https://webpack.js.org/concepts/module-federation/#dynamic-public-path

**Comment by rajeshbabu-oviva** [comment]
Thanks. Can confirm that setting publicPath to 'auto' on the remote application's webpack config fixed it for me.

**Comment by ChristopherHButler** [comment]
I am a bit late to the party but in my own Net project I have observed that running 'next build' does not not respect the publicPath property in the webpack config (next.config.js).

I have opened a bug report with next on this here: [https://github.com/vercel/next.js/issues/40558](https://github.com/vercel/next.js/issues/40558) and it's only an issue for me in production because of the way that I structure my files on aws s3. Regardless of what I am doing with it though, webpack should prepend the publicPath to script path's for example but it does not (even when I run next build locally.
